src/api/spaces/index.ts

```diff
@@ -52,30 +52,56 @@ export class SpacesAPI extends Request {
   public deleteSpace(params: IDeleteSpaceParams) {
     return this.request({
       errors: [UNAUTHORIZED, FORBIDDEN, DATA_SOURCE_UNAVAILABLE, NOT_FOUND],
-      url: '/:sourceKey/spaces/:id',
+      url: '/admin/:sourceKey/spaces/:id',
       method: 'DELETE',
       params: params
     });
   }
 
   /**
-   * Get space list.
+   * List all spaces (even if they are not visible by the current user).
    */
   public getSpaces(this: Request<ISpace[]>, params?: IDataSourceParams) {
     return this.request({
       errors: [UNAUTHORIZED, FORBIDDEN, DATA_SOURCE_UNAVAILABLE],
-      url: '/:sourceKey/spaces',
+      url: '/admin/:sourceKey/spaces',
       method: 'GET',
       params: params
     });
   }
 
   /**
-   * Get a space by id.
+   * Get a space by id (even if it is not visible by the current user).
    */
   public getSpace(this: Request<ISpace>, params: IGetSpaceParams) {
     return this.request({
       errors: [UNAUTHORIZED, FORBIDDEN, DATA_SOURCE_UNAVAILABLE, NOT_FOUND],
+      url: '/admin/:sourceKey/spaces/:id',
+      method: 'GET',
+      params: params
+    });
+  }
+}
+
+export class SpacesAPI extends Request {
+  /**
+   * List spaces visible by the current user.
+   */
+  public getSpaces(this: Request<ISpace[]>, params?: IDataSourceParams) {
```

Comment:
I am not convinced that we need to separate name spaces for admin and regular users.
As of today the only difference in use-case is that:
- An admin want to see all spaces when managing them
- Regular users want to see the spaces shared with them in order too access the visualizations shared with them.

Proposal:
- Name the APIs in a way that reflects the use-cases:
1. getAllSpaces
2. getSpacesSharedWithMe

Reply:
Let's do that, considering that me no longer have a "get space by id" operation. Having a distinct API for a single operation is definitely an overkill.